pkg/testing/tools/artifacts_api.go

```diff
@@ -310,9 +317,27 @@ func (aac ArtifactAPIClient) createAndPerformRequest(ctx context.Context, URL st
 		return nil, err
 	}
 
-	resp, err := aac.c.Do(req)
+	// Make the request with retries as the artifacts API can sometimes be flaky.
+	var resp *http.Response
+	// TODO (once we're on Go 1.22): replace with for numAttempts := range maxAttemptsForArtifactsAPICall {
+	for numAttempts := 0; numAttempts < maxAttemptsForArtifactsAPICall; numAttempts++ {
+		resp, err = aac.c.Do(req)
+		if err == nil {
```

Comment:
You should not continue to retry if the error is context.Cancelled or context.DeadlineExceeded.

Reply:
Fixed in adf118f6cd.